File: urbanairship-sdk/src/test/java/com/urbanairship/wallet/FieldTest.java
/* Copyright 2017 Urban Airship and Contributors */

package com.urbanairship.wallet;

import com.urbanairship.BaseTestCase;
import com.urbanairship.json.JsonMap;
import com.urbanairship.json.JsonValue;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class FieldTest extends BaseTestCase {

    @Test
    public void testFullField() throws Exception {
        String name = "name";
        String label = "label";
        Integer value = 5;

        Field field = Field.newBuilder()
                .setLabel(label)
                .setName(name)
                .setValue(value)
                .build();

        JsonValue expectedJsonValue = JsonMap.newBuilder()
                .put(label, label)
                .put("value", value)
                .build()
                .toJsonValue();

        assertEquals(name, field.getName());
        assertEquals(expectedJsonValue, field.toJsonValue());
    }

    @Test
    public void testFieldVariations() {
        Field.newBuilder()
                .setName("name")
                .setValue(5)
                .build();

        Field.newBuilder()
               .setName("name")
               .setLabel("label")
               .build();
    }

    @Test(expected = IllegalStateException.class)
    public void testFieldOnlyLabel() {
        Field.newBuilder()
              .setLabel("label")
              .build();
    }
}

File: urbanairship-core/src/test/java/com/urbanairship/wallet/FieldTest.kt
/* Copyright Airship and Contributors */
package com.urbanairship.wallet

import com.urbanairship.json.jsonMapOf
import org.junit.Assert
import org.junit.Test

public class FieldTest {

    @Test
    public fun testFullField() {
        val name = "name"
        val label = "label"
        val value = 5

        val field = Field.newBuilder()
            .setLabel(label)
            .setName(name)
            .setValue(value)
            .build()

        val expectedJsonValue = jsonMapOf(
            "label" to label,
            "value" to value
        ).toJsonValue()

        Assert.assertEquals(name, field.name)
        Assert.assertEquals(expectedJsonValue, field.toJsonValue())
    }

    @Test
    public fun testFieldVariations() {
        Field.newBuilder().setName("name").setValue(5).build()

        Field.newBuilder().setName("name").setLabel("label").build()
    }

    @Test(expected = IllegalStateException::class)
    public fun testFieldOnlyLabel() {
        Field.newBuilder().setLabel("label").build()
    }
}
